npm/pkg/dataplane/ipsets/ipsetmanager.go

```diff
@@ -174,12 +174,23 @@ func (iMgr *IPSetManager) AddToSets(addToSets []*IPSetMetadata, ip, podKey strin
 	iMgr.Lock()
 	defer iMgr.Unlock()
 
-	if err := iMgr.checkForIPUpdateErrors(addToSets, npmerrors.AppendIPSet); err != nil {
-		return err
+	// 1. check for errors and create any missing sets
+	for _, metadata := range addToSets {
+		prefixedName := metadata.GetPrefixName()
+		set, exists := iMgr.setMap[prefixedName]
+		if !exists {
+			// NOTE: any newly created IPSet will still be in the cache if an error is returned later
+			iMgr.createIPSet(metadata)
+			set = iMgr.setMap[prefixedName]
+		}
+		if set.Kind != HashSet {
+			return npmerrors.Errorf(npmerrors.AppendIPSet, false, fmt.Sprintf("ipset %s is not a hash set", prefixedName))
```

Comment:
Why the double loop here? we can put this in one giant loop ? At least some of the sets will have the right IPs if an error occurs.